File: preschool-management-backend/src/main/java/com/vusystem/preschool_management_backend/modules/mobile/service/DailyLogService.java
package com.vusystem.preschool_management_backend.modules.mobile.service;

import com.vusystem.preschool_management_backend.modules.mobile.dto.request.DailyLogBatchUpdateRequest;
import com.vusystem.preschool_management_backend.modules.mobile.dto.response.DailyLogResponse;

import com.vusystem.preschool_management_backend.modules.mobile.dto.response.DailyLogHistoryResponse;

import java.time.LocalDate;
import java.util.List;

public interface DailyLogService {
    
    List<DailyLogResponse> getDailyLogsForClass(Long classId, LocalDate date);

    void batchUpdateDailyLogs(LocalDate date, DailyLogBatchUpdateRequest request);

    // dùng cho luồng phụ huynh
    DailyLogResponse getDailyLogForChild(Long childId, LocalDate date);

    List<DailyLogHistoryResponse> getChildAttendanceHistory(Long childId, int year, int month);
}

File: mobile/src/services/dailyLog.service.ts
import { DailyLogResponse, DailyLogBatchUpdateRequest, DailyLogHistoryResponse } from '../types/dailyLog';

import axiosInstance from '../config/api.client';

export const dailyLogService = {
  getDailyLogsForClass: async (classId: number, date: string): Promise<DailyLogResponse[]> => {
    const response = await axiosInstance.get<{ status: number; message: string; data: DailyLogResponse[] }>(
      `/mobile/daily-logs/classes/${classId}`, 
      { params: { date } }
    );
    return response.data.data;
  },

  batchUpdateDailyLogs: async (date: string, request: DailyLogBatchUpdateRequest): Promise<void> => {
    await axiosInstance.put(
      `/mobile/daily-logs/batch`,
      request,
      { params: { date } }
    );
  },

  getDailyLogForChild: async (childId: number, date: string): Promise<DailyLogResponse | null> => {
    try {
      const response = await axiosInstance.get<{ status: number; message: string; data: DailyLogResponse }>(
        `/mobile/daily-logs/children/${childId}`,
        { params: { date } }
      );
      return response.data.data;
    } catch (error: any) {
      if (error.response?.status === 404) {
        return null;
      }
      throw error;
    }
  },

  getChildAttendanceHistory: async (childId: number, year: number, month: number): Promise<DailyLogHistoryResponse[]> => {
    const response = await axiosInstance.get<{ status: number; message: string; data: DailyLogHistoryResponse[] }>(
      `/mobile/daily-logs/history/children/${childId}`,
      { params: { year, month } }
    );
    return response.data.data;
  }
};
